osgp/platform/osgp-adapter-domain-tariffswitching/src/test/java/org/opensmartgridplatform/adapter/domain/tariffswitching/application/services/ScheduleManagementServiceTest.java

```diff
@@ -0,0 +1,137 @@
+import org.opensmartgridplatform.domain.core.repositories.SsldRepository;
+import org.opensmartgridplatform.domain.core.services.DeviceDomainService;
+import org.opensmartgridplatform.domain.core.services.OrganisationDomainService;
+import org.opensmartgridplatform.domain.core.valueobjects.LightValue;
+import org.opensmartgridplatform.domain.core.valueobjects.RelayType;
+import org.opensmartgridplatform.domain.core.valueobjects.ScheduleEntry;
+import org.opensmartgridplatform.shared.exceptionhandling.ComponentType;
+import org.opensmartgridplatform.shared.exceptionhandling.FunctionalException;
+import org.opensmartgridplatform.shared.infra.jms.CorrelationIds;
+import org.opensmartgridplatform.shared.infra.jms.RequestMessage;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+public class ScheduleManagementServiceTest{
+	@Mock
+	private DeviceDomainService deviceDomainService;
+	
+	@Mock
+	private DomainTariffSwitchingMapper domainCoreMapper;
+	
+	@Mock 
+	private SsldRepository ssldRepository;
+	
+	@Mock
+	private OrganisationDomainService organisationDomainService;
+	
+	@Mock
+    private OsgpCoreRequestMessageSender osgpCoreRequestMessageSender;
+
+	@Mock
+	private Device device;
+	
+	@Mock
+	private Ssld ssld;
+	
+	@Mock
+	private CorrelationIds correlationIds;
+	
+	@InjectMocks
+	private ScheduleManagementService scheduleManagementService;
+	
+	private List<ScheduleEntry> scheduleEntries;
+	private List<DeviceOutputSetting> deviceOutputSettings;
+	private List<LightValue> lightValues;
+	
+	
+	@BeforeEach
+	public void setup() throws FunctionalException {
+		this.scheduleEntries = new ArrayList<>();
+		this.deviceOutputSettings = new ArrayList<>();
+		this.lightValues = new ArrayList<>();
+		
+		final DeviceOutputSetting dos = Mockito.mock(DeviceOutputSetting.class);
+		when(dos.getOutputType()).thenReturn(RelayType.TARIFF_REVERSED);
+		
+		final ScheduleEntry entry = Mockito.mock(ScheduleEntry.class);
+		when(entry.getLightValue()).thenReturn(this.lightValues);
+		
+		final LightValue value = Mockito.mock(LightValue.class);
+		
+		//idk how many entries are typically required
+		for(int i =0; i < 3; i++) {
+			this.scheduleEntries.add(entry);
+			this.deviceOutputSettings.add(dos);
+			this.lightValues.add(value);
+		}
+		
+		when(this.device.getId()).thenReturn((long) 0.0f);
+		when(this.ssld.getOutputSettings()).thenReturn(this.deviceOutputSettings);
+		when(this.ssldRepository.findById(any())).thenReturn(Optional.of(this.ssld));
+		when(this.device.getIpAddress()).thenReturn("127.0.0.1");
+		when(this.deviceDomainService.searchActiveDevice(any(), any(ComponentType.class))).thenReturn(this.device);
+	}
+	
+	@Test
+	public void testSetTariffSchedule() throws FunctionalException {
+		when(this.device.getDeviceType()).thenReturn(Ssld.SSLD_TYPE);
+
+		final long scheduleTime = (long) 0.0f;
+		final String messageType = "testType";
+		final int priority = 1;
```

Comment:
Maybe introduce these as constants in the class, since they are also used (and tested) in testSetTariffScheduleFunctionalExceptionThrown()